tools/mo/openvino/tools/mo/pipeline/common.py

```diff
@@ -169,15 +169,17 @@ def convert_inputs_of_specific_ops(graph: Graph):
                                   ''.format(port_id, node.soft_get('name', node.id), precision))
                         in_port = node.in_port(port_id)
                         np_type = data_type_str_to_np(precision)
+                        in_node = node.in_port(port_id).get_source().node
+                        in_node_type = in_node.out_port(0).get_data_type()
+                        if np.issubdtype(in_node_type, np.integer) and np.issubdtype(np_type, np.integer):
+                            # do not convert if both source and destination types are of integer types
+                            # otherwise, it affects compatibility of MO IR Engine and TF FE
+                            # TF FE intents to use original model type for layers if it is possible.
+                            # Also, this skip is needed to get the same graphs from original IR and
+                            # restored with ir_reader_utils
+                            continue
```

Comment:
Actually, it can be not safe change. I remember Evgenya stopped to do it and proposed to apply only for `Const`.
@jane-intel, can you please advice if this is safe change? What pitfalls can be here?

Reply:
Discussed with @jane-intel. Skipping for non-const nodes should be done selectively only for conversions within int32, int64, uint32, uint64. Corrected current PR to do so.

But if we leave `int8`, `int4` (or similar) types on `Reshape, Squeeze, etc.` inputs potentially there can be problems, therefore for such cases current serializer behavior is left unchanged.